services/queue/src/utils.js

```diff
@@ -42,86 +41,108 @@ const artifactUtils = {
   async expire({ db, publicBucket, privateBucket, ignoreError, monitor, expires }) {
     let count = 0;
 
-    // Get all expired artifacts using a handler function, operating on one
-    // page of results at a time.  This does not use paginatedIterator because
-    // that would result in a much slower one-by-one expiration, instead of
-    // the parallel handling of each batch.
-    const getExpiredArtifacts = async (expires, handler) => {
-      let after_task_id_in = null;
-      let after_run_id_in = null;
-      let after_name_in = null;
-
-      while (true) {
-        const rows = await db.fns.get_queue_artifacts_paginated({
-          task_id_in: null,
-          run_id_in: null,
-          expires_in: expires,
-          page_size_in: 100,
-          after_task_id_in,
-          after_run_id_in,
-          after_name_in,
-        });
-
-        const entries = rows.map(exports.artifactUtils.fromDb);
-        await Promise.all(entries.map((item) => handler.call(item, item)));
-
-        if (!rows.length) {
-          break;
-        } else {
-          const lastRow = rows[rows.length - 1];
-          after_task_id_in = lastRow.task_id;
-          after_run_id_in = lastRow.run_id;
-          after_name_in = lastRow.name;
-        }
+    // Fetch all expired artifacts and batch delete the S3 ones
+    // then remove the entity from the database
+    // repeat until there are no more expired artifacts
+    while (true) {
+      const rows = await db.fns.get_expired_artifacts_for_deletion({
+        expires_in: expires,
+        page_size_in: 1000,
```

Comment:
on prod db table selecting 1000 records is performed under 200ms on average, so this should speed things up